architecture/README.md

```diff
@@ -0,0 +1,190 @@
+# Architecture
+Adds a Lifecycle aware ViewModel solution with support for Navigation and Observables. Kaluga favours an MVVM based architecture due to its clean delegation between UI and View State.
+
+## ViewModels
+ViewModels can be created by subclassing the `BaseViewModel` class. The ViewModel runs within its own coroutine scope that exists for the entire lifetime of the ViewModel.
+In addition ViewModels have their own lifecycle, allowing them to the paused and resumed by the view. When the ViewModel is resumed a `CoroutineScope` with a lifecycle limited to the resumed state is provided.
+Launch any coroutines within this scope to automatically cancel when the ViewModel is paused.
+
+```kotlin
+class SomeViewModel : BaseViewModel() {
+
+    init {
+        coroutineScope.launch {
+            // Do stuff that happens even when viewModel paused
+        }
+    }
+
+    override fun onResume(scope: CoroutineScope) {
+        super.onResume(scope)
+
+        scope.launch {
+            // Do stuff that happens only when viewmodel is active
+        } 
+    }
+
+}
+```
+
+On Android the AndroidX `ViewModel` serves as a base. The Viewmodel lifecycle matches onResume/onPause of the `Activity` or `Fragment` that created it.
+To achieve this, the View should be bound using the `KalugaViewModelLifecycleObserver.bind()` method.
+For convenience default implementations for `Activity`, `Fragment`, and `DialogFragment` exist (`KalugaViewModelActivity`, `KalugaViewModelFragment`, and `KalugaViewModelDialogFragment` respectively).
+Alternatively the user can call `didResume()` and `didPause()` on the ViewModel manually in the `onResume()` and `onPause()` methods.
+
+On iOS automatic setup was omitted due to ObjC classes requiring to be final.
+To use it, the user must manually call the ViewModels `didResume`/`didPause` in `viewDidAppear`/`viewDidDisappear`.
+
+## Observables
+Kaluga supports `Observables` (one way binding) and `Subjects` (Two way binding). An Object can be created through a `ReadOnlyProperty` (making it immutable on both sides), a `Flow` (allowing the flow to modify the observer), or a `BaseFlowable` (allowing both the Flow and the owner of BaseFlowable to modify the observer.
```

Comment:
> and `Subjects` (Two way binding)

Does Kaluga support stateless and stateful Subjects? 
Like  `PublishSubject` and `BehaviourSubject`, respectively, from RxSwiftt.

Reply:
Subjects are always stateful. LiveData, the basis for android, is stateful by design.

Reply:
So there is no way to have a two-way stateless bindings with Kaluga?

Reply:
Not using observables, since they are purely for data binding. Statelessness should be handled in the common code, where Flow/Channel will suffice.